src/App.tsx

```diff
@@ -1,61 +1,161 @@
+/* eslint-disable default-case */
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userArgument: number | string): User | null {
+  const foundUserById
+  = usersFromServer.find(user => user.id === userArgument);
+
+  const foundUserByName
+  = usersFromServer.find(user => user.name === userArgument);
+
+  switch (typeof userArgument) {
+    case 'string':
+      return foundUserByName || null;
+    default:
+      return foundUserById || null;
+  }
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [titleName, titleState] = useState('');
+  const [userName, userState] = useState('');
+  const [titleBoolean, titleError] = useState(false);
+  const [userBoolean, userError] = useState(false);
+  const [todosArray, todoArrayModify] = useState(todos);
+
+  const handleFormData = () => {
+    const regexChars = /[\wа-я\s]/ig;
+    const titleModified: string | undefined
+    = titleName.match(regexChars)?.join('');
+    const currentUser = getUser(userName);
+    const id = currentUser?.id;
+    const todosIdCollection = todosArray.map(todo => todo.id);
+    const todosIdMaxValue = Math.max(...todosIdCollection);
+    const newTodo: Todo | undefined = {
+      id: todosIdMaxValue + 1,
+      userId: id,
+      title: titleModified,
+      completed: false,
+      user: currentUser,
+    };
+
+    switch (true) {
+      case !titleModified
+      && !userName:
+        titleError(true);
+        userError(true);
+        break;
+      case !titleModified:
+        titleError(true);
+        userError(false);
+        break;
+      case !userName:
+        userError(true);
+        titleError(false);
+        break;
+      case !!titleModified
+      && !!userName:
+        todoArrayModify(current => [...current, newTodo]);
+        titleError(false);
+        userError(false);
+        titleState('');
+        userState('');
+        break;
+    }
```

Comment:
1. `switch (true)` - bad case? then better to use if
2. You have repeatable conditions. Divide them. Ex: `titleError` has `false` value as default and `true` only if we haven't title and clicked on `Add` button